### DISCUSSION TITLE
Unable to set visibility for file minified chmod(): Operation not permitted

### DISCUSSION THREAD

**Discussion by marcoonesto** [pimcore/pimcore#16367] [Q&A / Support]
Hello everyone,
I have a dev system VM with Debian 12 working with pimcore 11 in a azure vm, mariadb saas and azure blob storage configured with flysystem.yml
We were trying to create a production environment with the same configurations from dev but when installing the azure connector and configuring the flysystem.yml and trying to login to backend https://mysite/admin I receive this error
![image](https://github.com/pimcore/pimcore/assets/109591840/5b242d64-6b4b-4991-9b4e-d5c0b227b853)

`An exception has been thrown during the rendering of a template ("Unable to set visibility for file minified_javascript_core_47a54269fc979213ee28bd8382de714f.js. chmod(): Operation not permitted").`

In addition from browser I can read this 
```
League\Flysystem\UnableToSetVisibility:
Unable to set visibility for file minified_javascript_core_47a54269fc979213ee28bd8382de714f.js. chmod(): Operation not permitted
```

I'll paste the full error log from prod.log in /pimcore/var/log/prod.log

`[2023-12-14T11:25:37.505999+00:00] request.CRITICAL: Uncaught PHP Exception Twig\Error\RuntimeError: "An exception has been thrown during the rendering of a template ("Unable to set visibility for file minified_javascript_core_47a54269fc979213ee28bd8382de714f.js. chmod(): Operation not permitted")." at /usr/share/nginx/pimcore/vendor/pimcore/admin-ui-classic-bundle/templates/admin/index/index.html.twig line 632 {"exception":"[object] (Twig\\Error\\RuntimeError(code: 0): An exception has been thrown during the rendering of a template (\"Unable to set visibility for file minified_javascript_core_47a54269fc979213ee28bd8382de714f.js. chmod(): Operation not permitted\"). at /usr/share/nginx/pimcore/vendor/pimcore/admin-ui-classic-bundle/templates/admin/index/index.html.twig:632)\n[previous exception] [object] (League\\Flysystem\\UnableToSetVisibility(code: 0): Unable to set visibility for file minified_javascript_core_47a54269fc979213ee28bd8382de714f.js. chmod(): Operation not permitted at /usr/share/nginx/pimcore/vendor/league/flysystem/src/UnableToSetVisibility.php:33)"} []`

All permissions are correct and verified, I also tried to give 777 to all pimcore's folder

**Comment by symos** [comment]
Did you ever find a solution to this? We also have this problem. We have to delete the contents of the var/admin folder in order to bypass this, but this is obviously not a proper solution.

**Comment by faiyazalam** [comment]
@symos 

The `var/admin/` directory needs to be writable by both the server user and server group. To fix this, I ran:

`chown -R www-data:www-data var/admin/
`

This successfully resolved the permission issue.

For a more complete solution, you should set proper ownership for all relevant directories:

`chown -R YOUR_USER:YOUR_GROUP var/ public/var/
`

For detailed guidance on file permissions, please refer to the official documentation:
https://docs.pimcore.com/platform/Pimcore/Installation_and_Upgrade/System_Setup_and_Hosting/File_Permissions
